### PR TITLE
remove deprecated G8 methods while keeping G7 compatibility

### PR DESCRIPTION
**Pull request by doudou**
@joaobrittoneto I've done what I can to remove all the deprecation warnings from G8 while keeping G7 compatibility (both should compile fine). However, I don't have an environment to test it right now. Do you think you could test it ?

### CODE REVIEW

**Comment by joaobrittoneto** [comment]
I'll test it for G7 (the one I have).

**Comment by doudou** [comment]
> I'll test it for G7 (the one I have).

Great ! Thanks !

**Comment by joaobrittoneto** [comment]
First problem I'm having is with [gazebo_thruster](https://github.com/rock-gazebo/simulation-gazebo_thruster). 
With the compensated force thing, `gazebo_thruster` is bonded to `gazebo_underwater` (Vector6, Matrix6 for the math and .proto).
The math in `gazebo_thruster` needs update as well.

Edit:
['gazebo_thruster'](https://github.com/Brazilian-Institute-of-Robotics/simulation-gazebo_thruster/pull/11#pullrequestreview-47897509) update

**Comment by joaobrittoneto** [comment]
Second problem happens when I try to run the simulation with `dive sim tank`.

>[Err] [World.cc:1468] World[tank] is attempting to load a plugin, but detected an incorrect plugin type. Plugin filename[libgazebo_underwater.so] name[underwater]
gzserver: /usr/include/boost/smart_ptr/shared_ptr.hpp:648: typename boost::detail::sp_member_access<T>::type boost::shared_ptr<T>::operator->() const [with T = gazebo::transport::Node; typename boost::detail::sp_member_access<T>::type = gazebo::transport::Node*]: Assertion `px != 0' failed.

**Comment by doudou** [comment]
Thanks ... I was optimistically hoping some more straightforward result. I think it will have to wait until I have the need to make an underwater scene unfortunately.

**Comment by joaobrittoneto** [comment]
I think the last error is something related with those commits, I think.
https://github.com/rock-gazebo/simulation-gazebo_underwater/commit/252f9139b795aface3824198a13d476990f0702f, https://github.com/rock-gazebo/simulation-gazebo_underwater/commit/6ccf104591cbaa6fd20143ab97565c5e00bc4434.
I also happens on master. It seems that I need to update the sdf file as well.

Edit:
After moving the `gazebo_underwater` from the world sdf to the model sdf it worked.

**Comment by joaobrittoneto** [comment]
The CoG is computed wrong. The vehicle is leaning to the left side.
It computes the CoG at:
>[Msg] GazeboUnderwater: Model's CoG: (0.000181 2e-05 -9.3e-05) meters

while it should be at:
>[Msg] GazeboUnderwater: Model's CoG: (0.000181 0.020006 -9.3e-05) meters

File: src/GazeboUnderwater.cpp [hunk 1]
```diff
@@ -77,16 +78,15 @@ namespace gazebo_underwater
...
         for (physics::Link_V::iterator it = links.begin(); it != links.end(); ++it)
             if(!(*it)->GetKinematic())
             {
                 // Set Inertial's CoG related with the parent link
-                temp = *(*it)->GetInertial();
-                math::Pose pose = (*it)->GetRelativePose();
-                pose.pos += pose.rot.RotateVector(temp.GetCoG());
+                Inertial temp = *(*it)->GetInertial();
+                Pose3d pose = GzGetIgn((**it), RelativePose, ());
+                pose.CoordPositionAdd(pose.Rot().RotateVector(GzGetIgn(temp, CoG, ())));
```
Comment (joaobrittoneto) [inline src/GazeboUnderwater.cpp]: The `CoordPositionAdd` does not do the same [thing](https://bitbucket.org/osrf/gazebo/src/a29d8dabed9a14fb6bafdcd31c84b160c00b127c/gazebo/math/Pose.cc?at=gazebo7_7.0.0&fileviewer=file-view-default#Pose.cc-164) as the previous.

**Comment by doudou** [comment]
OK. How do you think we should move forward with this ?

We could merge the two PRs, since you've ironed out the bugs. You guys still have the option of sticking with the BIR repositories until you migrate to the model-based plugin.

Or we could remove the commit that makes it a model plugin and put that one on a PR.

I would prefer the first solution personally, but I don't have any running system with that stuff, so I don't really have any "skin in the game" ...

**Comment by joaobrittoneto** [comment]
>I would prefer the first solution personally

I agree. I tested and it worked with the last update in the [thruster plugin](https://github.com/rock-gazebo/simulation-gazebo_thruster/pull/1).
It makes more sense to have the parameters defined in the model's sdf. It would allows a world's sdf with multiples robot to not be crowded.

The only point I'm not  sure how to handle is the `fluid_velocity`. This one I think would make more sense to be defined in the world's sdf. Multiples robots would experience the same current, but it is a minor thing that can be dealt latter. 

By the way, how to have multiples robots in the same simulation??

**Comment by doudou** [comment]
> By the way, how to have multiples robots in the same simulation??

What do you see as a potential problem ?

**Comment by doudou** [comment]
> I agree. I tested and it worked with the last update in the thruster plugin.

OK. I'll merge both then.

**Pull request merged**